_posts/2023-06-20-introducing-iam-for-openfaas.md

```diff
@@ -0,0 +1,479 @@
+---
+title: "Introducing Identity and Access Management (IAM) for OpenFaaS"
+description: "Granularly control access to the OpenFaaS REST API with IAM for OpenFaaS"
+date: 2023-06-20
+categories:
+- sso
+- authorization
+- iam
+- oauth
+author_staff_member: han
+---
+
+## Get started with IAM for OpenFaaS
+
+In this tutorial we are going to walk you through all the steps required to deploy OpenFaaS with Identity and Access Management (IAM) enabled.
+
+In the next sections we are going to:
+
+- Setup ingress and all other prerequisites to deploy OpenFaaS.
+- Configure your identity provider and register it with OpenFaaS.
+- Create roles and policies for fine grained access to the OpenFaaS API.
+- Configure and deploy the OpenFaaS dashboard.
+- Authenticate and manage functions using the faas-cli
+
+### Deploy OpenFaaS
+
+Create the namespaces for OpenFaaS and its functions:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
+```
+
+Create a secret for your OpenFaaS license:
+
+```bash
+kubectl create secret generic \
+  -n openfaas \
+  openfaas-license \
+  --from-file license=$HOME/.openfaas/LICENSE
+```
+
+Add the OpenFaaS helm chart repo:
+
+```
+helm repo add openfaas https://openfaas.github.io/faas-netes/
+```
+
+Configure ingress to make the OpenFaaS gateway accessible to the users.
+
+The below instructions show how to set up Ingress with a TLS certificate using Ingress Nginx. You can also use any other ingress-controller, inlets-pro or an Istio Gateway.
+
+Install [cert-manager](https://cert-manager.io/docs/), which is used to manage TLS certificates.
+
+You can use Helm, or [arkade](https://github.com/alexellis/arkade):
+
+```bash
+arkade install cert-manager
+```
+
+Install ingress-nginx using arkade or Helm:
+
+```bash
+arkade install ingress-nginx
+```
+
+Create an ACME certificate issuer:
+
+```bash
+export EMAIL="mail@example.com"
+
+cat > issuer-prod.yaml <<EOF
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: $EMAIL
+    privateKeySecretRef:
+      name: letsencrypt
+    solvers:
+    - http01:
+        ingress:
+          class: nginx
+EOF
+```
+
+```bash
+kubectl apply -f issuer-prod.yaml
+```
+
+Create an ingress record for the gateway:
+
+```bash
```

Comment:
I think it makes sense to do the dashboard ingress at the start along with the gateway ingress. Otherwise they may create DNS records twice.

Add to the pre-reqs that they'll need to create two DNS entries, and that these can either be on the public Internet *or kept within the internal network*.

Reply:
Moved the dashboard ingress and added a note to prepare the DNS.